overlord/hookstate/hookmgr.go

```diff
@@ -270,9 +274,27 @@ func snapCmd() string {
 	return filepath.Join(filepath.Dir(exe), "../../bin/snap")
 }
 
-func runHookAndWait(snapName string, revision snap.Revision, hookName, hookContext string, tomb *tomb.Tomb) ([]byte, error) {
+var syscallKill = syscall.Kill
+var cmdWaitTimeout = 5 * time.Second
+
+func killemAll(cmd *exec.Cmd) error {
+	pgid, err := syscall.Getpgid(cmd.Process.Pid)
```

Comment:
We should avoid using the syscall package when there are proper high-level abstractions out of it. cmd.Process itself should have a proper API for what is being done below:

https://golang.org/pkg/os/#Process